test/test_build.py

```diff
@@ -0,0 +1,280 @@
+import os
+import json
+from pathlib import Path
+import subprocess
+from unittest import TestCase, mock
+from biggerquery.cli import walk_module_files
+from biggerquery.version import get_version
+from biggerquery.build import now, get_docker_image_id, build_docker_image_tag, \
+    clear_image_leftovers, clear_package_leftovers, clear_dags_leftovers, auto_configuration, \
+    get_docker_repository_from_deployment_config, project_setup, secure_get_version
+from example_project.project_setup import DOCKER_REPOSITORY, PROJECT_NAME
+
+TEST_PROJECT_PATH = Path(__file__).parent / 'example_project'
+IMAGE_DIR_PATH = TEST_PROJECT_PATH / 'image'
+DAGS_DIR_PATH = TEST_PROJECT_PATH / '.dags'
+DIST_DIR_PATH = TEST_PROJECT_PATH / 'dist'
+EGGS_DIR_PATH = TEST_PROJECT_PATH / f'{PROJECT_NAME}.egg-info'
+BUILD_PATH = TEST_PROJECT_PATH / 'build'
+
+
+def resolve(path: Path):
+    return str(path.absolute())
+
+
+class TestProject:
+    def run_build(self, cmd: str):
+        output = subprocess.getoutput(f'cd {resolve(TEST_PROJECT_PATH)};{cmd}')
+        print(output)
+        return output
+
+
+def mkdir(dir_path: Path):
+    if not os.path.isdir(dir_path):
+        os.mkdir(dir_path)
+
+
+def create_image_leftovers(test_project_dir_path: Path = TEST_PROJECT_PATH):
+    mkdir(test_project_dir_path / 'image')
+    (test_project_dir_path / 'image' / 'leftover').touch()
+
+
+def create_package_leftovers(
+        test_project_dir_path: Path = TEST_PROJECT_PATH,
+        project_name: str = PROJECT_NAME):
+    mkdir(test_project_dir_path / 'build')
+    (test_project_dir_path / 'build' / 'leftover').touch()
```

Comment:
could we move lines like this to mkdir method?

Reply:
The `touch` method creates a file, so we can't